staging/src/k8s.io/component-helpers/resource/helpers.go

```diff
@@ -56,14 +58,14 @@ type PodResourcesOptions struct {
 var supportedPodLevelResources = sets.New(v1.ResourceCPU, v1.ResourceMemory)
 
 func SupportedPodLevelResources() sets.Set[v1.ResourceName] {
-	return supportedPodLevelResources
+	return supportedPodLevelResources.Clone().Insert(v1.ResourceHugePagesPrefix)
 }
 
 // IsSupportedPodLevelResources checks if a given resource is supported by pod-level
 // resource management through the PodLevelResources feature. Returns true if
 // the resource is supported.
 func IsSupportedPodLevelResource(name v1.ResourceName) bool {
-	return supportedPodLevelResources.Has(name)
+	return supportedPodLevelResources.Has(name) || strings.HasPrefix(string(name), v1.ResourceHugePagesPrefix)
```

Comment:
a couple comments here:
1. this is effectively relaxing validation, which means we have to make sure it isn't active by default in the first release it is present in (1.33 will choke on pods this allows in with pod-level hugepages requests/limits)
2. this allowed things into the API which weren't actually implemented or tested yet (implementation is in https://github.com/kubernetes/kubernetes/pull/131089) ... we generally want to make sure implementation merges with API

both of those issues need to be addressed before pod-level resources enables by default

Reply:
oh, I just realized this merged in 1.33, not 1.34. whew

Reply:
Kevin also clarified that this PR did implement support. The implementation is implicit - pod level huge pages were already set on the pod cgroup using the Pod{Requests,Limits} helper which already factors in pod-level resources, so once we allow huge-pages to be set at the pod-level, the implementation is already there.